### PR TITLE
#165442657 Refactor transaction endpoints to use database

### PR DESCRIPTION
**Pull request by fxola**
#### What does this PR do?
- Refactor transaction endpoints to use database

#### Description of Task to be completed?
- update transaction model with a create method for storing new
transactions in the database
- refactor transaction controller and service methods to use async await
- update transactions failing test to pass
- update error handler controller on entry file
- add .env.example file for config details
- finishes[#165442657]

#### How should this be manually tested?
- Clone the repository using ```git clone https://github.com/fxola/Banka.git .``` in your terminal
- run ```npm install``` on your terminal
- run ```npm run test``` to see test results

#### Any background context you want to provide?
- See the env.example file for configuration details

#### What are the relevant pivotal tracker stories?
- PT Story link - [#165442657] (https://www.pivotaltracker.com/n/projects/2321791/stories/165442657)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A

### CODE REVIEW

File: server/src/services/transaction.service.js [hunk 1]
```diff
@@ -20,10 +20,8 @@ class TransactionService {
    * @returns {object} API response object
    * @memberof TransactionService
    */
-  static makeTransaction(accountNumber, amount, cashier, transactionType) {
-    const foundAccount = mockData.accounts.find(
-      account => account.accountNumber === parseInt(accountNumber, 10)
-    );
+  static async makeTransaction(accountNumber, amount, cashier, transactionType) {
+    const foundAccount = await Account.find(accountNumber);
```
Comment (ajimae) [inline server/src/services/transaction.service.js]: what are you trying to achieve here?

Comment (fxola) [reply server/src/services/transaction.service.js]: i'm trying to get the details of the account that matches the account number

Comment (fxola) [reply server/src/services/transaction.service.js]: so that i can perform any transaction on the account gotten

**Review by ajimae — COMMENTED** [review]
Hi Abass, kindly look at the comment I left on this PR.
Cheers.

File: server/src/services/transaction.service.js (hunk 1 above)
Comment (fxola) [reply server/src/services/transaction.service.js]: i'm trying to get the account details of the account that matches that account number

> Hi Abass, kindly look at the comment I left on this PR.
> Cheers.

**Pull request merged**